frontend/packages/knative-plugin/src/utils/create-eventsources-utils.ts

```diff
@@ -90,58 +75,36 @@ export const getEventSourceData = (source: string) => {
   return eventSourceData[source];
 };
 
-export const useKnativeEventingAccess = (model: K8sKind, namespace: string): boolean => {
-  const canCreateEventSource = useAccessReview({
+export const getKnativeEventingAccess = async (model: K8sKind, namespace: string) => {
+  const canCreateEventSource = await checkAccess({
     group: model.apiGroup,
     resource: model.plural,
     namespace,
     verb: 'create',
   });
-  return canCreateEventSource;
+  return canCreateEventSource?.status?.allowed;
 };
 
-export const useEventSourceList = (namespace: string) => {
-  const eventSourceList = {
-    ...(useKnativeEventingAccess(EventSourceCronJobModel, namespace) && {
-      [EventSourceCronJobModel.kind]: {
-        name: EventSourceCronJobModel.kind,
-        iconUrl: cronJobSourceImg,
-        displayName: EventSourceCronJobModel.label,
-        title: EventSourceCronJobModel.kind,
-      },
-    }),
-    ...(useKnativeEventingAccess(EventSourceSinkBindingModel, namespace) && {
-      [EventSourceSinkBindingModel.kind]: {
-        name: EventSourceSinkBindingModel.kind,
-        iconUrl: containerSourceImg,
-        displayName: EventSourceSinkBindingModel.label,
-        title: EventSourceSinkBindingModel.kind,
-      },
-    }),
-    ...(useKnativeEventingAccess(EventSourceApiServerModel, namespace) && {
-      [EventSourceApiServerModel.kind]: {
-        name: EventSourceApiServerModel.kind,
-        iconUrl: apiServerSourceImg,
-        displayName: EventSourceApiServerModel.label,
-        title: EventSourceApiServerModel.kind,
-      },
-    }),
-    ...(useKnativeEventingAccess(EventSourceKafkaModel, namespace) && {
-      [EventSourceKafkaModel.kind]: {
-        name: EventSourceKafkaModel.kind,
-        iconUrl: kafkaSourceImg,
-        displayName: EventSourceKafkaModel.label,
-        title: EventSourceKafkaModel.kind,
-      },
-    }),
-    ...(useKnativeEventingAccess(EventSourceCamelModel, namespace) && {
-      [EventSourceCamelModel.kind]: {
-        name: EventSourceCamelModel.kind,
-        iconUrl: camelSourceImg,
-        displayName: EventSourceCamelModel.label,
-        title: EventSourceCamelModel.kind,
-      },
-    }),
-  };
+export const getEventSourceList = (
+  eventSourceModelList: K8sKind[],
+  namespace: string,
+): NormalizedEventSources => {
+  const eventSourceList = _.reduce(
+    eventSourceModelList,
+    (accumulator, eventSourceModel) => {
+      return {
+        ...accumulator,
+        ...(getKnativeEventingAccess(eventSourceModel, namespace) && {
```

Comment:
Checking access should be done via hooks in the component.
Furthermore this function returns a promise but it used inside a non-async function.

Reply:
can make it `Async` function , earlier have made use if `useAccessReview` but as per hooks rule it can only be used in a hook or component and here we have list of event sources models so used reducer to handle it with `checkAccess`.